dartsim/src/SDFFeatures_TEST.cc

```diff
@@ -95,16 +109,196 @@ World LoadWorld(const std::string &_world)
+    }
+    else
+    {
+      newModel = world->ConstructModel(newSdfModel);
+    }
+
+    EXPECT_NE(nullptr, newModel);
+    if (nullptr != newModel)
+    {
+      modelMap[_model] = newModel;
+    }
+  };
+
+  for (uint64_t i = 0; i < sdfWorld->ModelCount(); ++i)
+  {
+    const auto *model = sdfWorld->ModelByIndex(i);
+    createModel(model);
+    for (uint64_t nestedInd = 0; nestedInd < model->ModelCount(); ++nestedInd)
+    {
+      createModel(model->ModelByIndex(nestedInd), model);
+    }
+  }
+
+  for (auto [sdfModel, physModel] : modelMap)
+  {
+    for (uint64_t li = 0; li < sdfModel->LinkCount(); ++li)
+    {
+      const auto *link = sdfModel->LinkByIndex(li);
+      sdf::Link newSdfLink;
+      newSdfLink.SetName(link->Name());
+      newSdfLink.SetRawPose(ResolveSdfPose(link->SemanticPose()));
+      newSdfLink.SetInertial(link->Inertial());
+
+      auto newLink = physModel->ConstructLink(newSdfLink);
+      EXPECT_NE(nullptr, newLink);
+      if (nullptr == newLink)
+        return nullptr;
+
+      linkMap[link] = newLink;
+    }
+  }
+
+  for (auto [sdfLink, physLink] : linkMap)
+  {
+    for (uint64_t ci = 0; ci < sdfLink->CollisionCount(); ++ci)
+    {
+      physLink->ConstructCollision(*sdfLink->CollisionByIndex(ci));
+    }
+  }
+
+  for (auto [sdfModel, physModel] : modelMap)
+  {
+    for (uint64_t ji = 0; ji < sdfModel->JointCount(); ++ji)
+    {
+      const auto *sdfJoint = sdfModel->JointByIndex(ji);
+      std::string resolvedParentLinkName;
+      const auto resolveParentErrors =
+          sdfJoint->ResolveParentLink(resolvedParentLinkName);
+      EXPECT_TRUE(resolveParentErrors.empty()) << resolveParentErrors;
+
+      std::string resolvedChildLinkName;
+      const auto resolveChildErrors =
+        sdfJoint->ResolveChildLink(resolvedChildLinkName);
+      EXPECT_TRUE (resolveChildErrors.empty()) << resolveChildErrors;
+
+      sdf::Joint newSdfJoint;
+      newSdfJoint.SetName(sdfJoint->Name());
+      if (sdfJoint->Axis(0))
+      {
+        newSdfJoint.SetAxis(0, ResolveJointAxis(*sdfJoint->Axis(0)));
+      }
+      if (sdfJoint->Axis(1))
+      {
+        newSdfJoint.SetAxis(1, ResolveJointAxis(*sdfJoint->Axis(1)));
+      }
+      newSdfJoint.SetType(sdfJoint->Type());
+      newSdfJoint.SetRawPose(ResolveSdfPose(sdfJoint->SemanticPose()));
+      newSdfJoint.SetThreadPitch(sdfJoint->ThreadPitch());
+
+      newSdfJoint.SetParentLinkName(resolvedParentLinkName);
+      newSdfJoint.SetChildLinkName(resolvedChildLinkName);
+
+      physModel->ConstructJoint(newSdfJoint);
+    }
+  }
+
+  return world;
+}
+/////////////////////////////////////////////////
+WorldPtr LoadWorld(const std::string &_world,
+                   LoaderType _loader = LoaderType::Whole)
+{
+  switch(_loader)
+  {
+    case LoaderType::Whole:
+      return LoadWorldWhole(_world);
+    case LoaderType::Piecemeal:
+      return LoadWorldPiecemeal(_world);
+    default:
+      return LoadWorldWhole(_world);
```

Comment:
Should we add a warning message here that `LoaderType` was neither `Whole` or `Piecemeal`? This may be useful information to have down the road in case we add on to these tests and/or something fails in CI.

Reply:
Added a warning in https://github.com/ignitionrobotics/ign-physics/pull/220/commits/53ce2b79c4059185820f7d5ded29221a9cabcb2b